packages/tangle/scheduler.go

```diff
@@ -135,21 +131,23 @@ func (s *Scheduler) Setup() {
 			}
 		}
 
-		s.updateDependencies(messageID, s.tryReady(messageID))
+		s.tryReady(messageID)
 	}))
 
-	onMessageScheduled := func(messageID MessageID) { s.updateDependencies(messageID, nil) }
-	s.tangle.Scheduler.Events.MessageScheduled.Attach(events.NewClosure(onMessageScheduled))
+	s.tangle.Scheduler.Events.MessageScheduled.Attach(events.NewClosure(s.updateApprovers))
 
 	onMessageConfirmed := func(messageID MessageID) {
 		var scheduled bool
 		s.tangle.Storage.MessageMetadata(messageID).Consume(func(messageMetadata *MessageMetadata) {
-			scheduled = messageMetadata.Scheduled() || messageMetadata.ScheduledBypass() // TODO: remove bypass check
+			scheduled = messageMetadata.Scheduled() || messageMetadata.ScheduledBypass() // Issue #1855 TODO: remove bypass check
 		})
 		if scheduled {
 			return
 		}
-		onMessageScheduled(messageID)
+		s.tangle.Storage.Message(messageID).Consume(func(message *Message) {
+			s.unsubmit(message)
```

Comment:
If a message has been confirmed and not scheduled yet but about to be marked as ready, it should just be unsubmitted as it would be filter out from the dispatcher anyway